diff --git a/node_modules/react-native/Libraries/BatchedBridge/MessageQueue.js b/node_modules/react-native/Libraries/BatchedBridge/MessageQueue.js
index 298d2e9..b0bbb99 100644
--- a/node_modules/react-native/Libraries/BatchedBridge/MessageQueue.js
+++ b/node_modules/react-native/Libraries/BatchedBridge/MessageQueue.js
@@ -412,6 +412,7 @@ class MessageQueue {
       }
       const moduleMethods = this.getCallableModule(module);
       if (!moduleMethods) {
+        return; /* iOS26_MQ_PATCHED */
         const callableModuleNames = Object.keys(this._lazyCallableModules);
         const n = callableModuleNames.length;
         const callableModuleNameList = callableModuleNames.join(', ');
diff --git a/node_modules/react-native/ReactCommon/react/runtime/ReactInstance.cpp b/node_modules/react-native/ReactCommon/react/runtime/ReactInstance.cpp
index 29d2910..2e364a2 100644
--- a/node_modules/react-native/ReactCommon/react/runtime/ReactInstance.cpp
+++ b/node_modules/react-native/ReactCommon/react/runtime/ReactInstance.cpp
@@ -293,22 +293,8 @@ void ReactInstance::callFunctionOnModule(
         "methodName",
         methodName);
     auto it = callableModules_.find(moduleName);
-    if (it == callableModules_.end()) {
-      std::ostringstream knownModules;
-      int i = 0;
-      for (it = callableModules_.begin(); it != callableModules_.end();
-           it++, i++) {
-        const char* space = (i > 0 ? ", " : " ");
-        knownModules << space << it->first;
-      }
-      throw jsi::JSError(
-          runtime,
-          "Failed to call into JavaScript module method " + moduleName + "." +
-              methodName +
-              "(). Module has not been registered as callable. Registered callable JavaScript modules (n = " +
-              std::to_string(callableModules_.size()) +
-              "):" + knownModules.str() +
-              ". Did you forget to call `registerCallableModule`?");
+    if (it == callableModules_.end()) { /* iOS26_MQ_PATCHED */
+      return;
     }
 
     if (std::holds_alternative<jsi::Function>(it->second)) {
diff --git a/node_modules/react-native/scripts/codegen/generate-artifacts-executor/generateRCTThirdPartyComponents.js b/node_modules/react-native/scripts/codegen/generate-artifacts-executor/generateRCTThirdPartyComponents.js
index fdd7e95..8d77992 100644
--- a/node_modules/react-native/scripts/codegen/generate-artifacts-executor/generateRCTThirdPartyComponents.js
+++ b/node_modules/react-native/scripts/codegen/generate-artifacts-executor/generateRCTThirdPartyComponents.js
@@ -138,7 +138,7 @@ function generateRCTThirdPartyComponents(
     .flatMap(library => {
       const components = componentsInLibraries[library];
       return components.map(({componentName, className}) => {
-        return `\t\t@"${componentName}": NSClassFromString(@"${className}"), // ${library}`;
+        return `\t\t{ Class cls = NSClassFromString(@"${className}"); if (cls) components[@"${componentName}"] = cls; } // ${library}`;
       });
     })
     .join('\n');
diff --git a/node_modules/react-native/scripts/codegen/templates/RCTThirdPartyComponentsProviderMM.template b/node_modules/react-native/scripts/codegen/templates/RCTThirdPartyComponentsProviderMM.template
index 4ce5107..1d2dd2b 100644
--- a/node_modules/react-native/scripts/codegen/templates/RCTThirdPartyComponentsProviderMM.template
+++ b/node_modules/react-native/scripts/codegen/templates/RCTThirdPartyComponentsProviderMM.template
@@ -19,9 +19,9 @@
   static dispatch_once_t nativeComponentsToken;
 
   dispatch_once(&nativeComponentsToken, ^{
-    thirdPartyComponents = @{
+    NSMutableDictionary<NSString *, Class<RCTComponentViewProtocol>> *components = [NSMutableDictionary new];
 {thirdPartyComponentsMapping}
-    };
+    thirdPartyComponents = [components copy];
   });
 
   return thirdPartyComponents;
