diff --git a/node_modules/react-native-google-mobile-ads/ios/RNGoogleMobileAds/RNGoogleMobileAdsModule.mm b/node_modules/react-native-google-mobile-ads/ios/RNGoogleMobileAds/RNGoogleMobileAdsModule.mm
index 776bdaf..d1bb7b2 100644
--- a/node_modules/react-native-google-mobile-ads/ios/RNGoogleMobileAds/RNGoogleMobileAdsModule.mm
+++ b/node_modules/react-native-google-mobile-ads/ios/RNGoogleMobileAds/RNGoogleMobileAdsModule.mm
@@ -176,15 +176,17 @@ RCT_EXPORT_METHOD(setAppMuted : (BOOL *)muted) {
 }
 
 - (NSDictionary *)constantsToExport {
+#if !TARGET_OS_MACCATALYST
   return @{
   // Precision types in ad revenue events.
   // See: https://developers.google.com/admob/ios/impression-level-ad-revenue#objective-c
-#if !TARGET_OS_MACCATALYST
     @"REVENUE_PRECISION_UNKNOWN" : @(GADAdValuePrecisionUnknown),
     @"REVENUE_PRECISION_ESTIMATED" : @(GADAdValuePrecisionEstimated),
     @"REVENUE_PRECISION_PUBLISHER_PROVIDED" : @(GADAdValuePrecisionPublisherProvided),
     @"REVENUE_PRECISION_PRECISE" : @(GADAdValuePrecisionPrecise)
   };
+#else
+  return @{};
 #endif
 }
 
